client/client.go

```diff
@@ -349,11 +349,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
-		<-done
-		if err == nil {
-			err = ctx.Err()
+		if err = resp.Body.Close(); err != nil {
```

Comment:
I think this is correct. If close fails and reading routine is blocked, we have no way to stop the reading routine anyway.

We break out here when it meets any error, so we could know that `err == nil` before waiting for reading routine done. And reading routine will always set err to `use of closed connection` error(which is caused by close body), and we could reset it to context error.

Another way is to compare err with `use of closed connection` error to check whether it is caused by body close. But there is no standard way to check this error, and it is tricky to compare string directly.